app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -54,21 +54,22 @@
 
     @NonNull
     @Override
-    public Completable querySearch(@NonNull final String query) {
-        return freeSoundApiService.search(get(query))
-                                  .map(DefaultSearchDataModel::toResults)
-                                  .doOnSubscribe(__ -> searchStateOnceAndStream
-                                          .onNext(SearchState.inProgress()))
-                                  .doOnSuccess(this::storeValueAndClearError)
-                                  .doOnError(storeError(query))
-                                  .toCompletable()
-                                  .onErrorComplete();
+    public Completable querySearch(@NonNull final String query,
+                                   @NonNull final Completable afterThis) {
```

Comment:
Could you rename this to maybe preliminary or something similar? afterThis seems a bit weird in the chain below.